query-node/mappings/storage/index.ts

```diff
@@ -250,8 +303,32 @@ export async function storage_DataObjectsMoved({ event, store }: EventContext &
 
 export async function storage_DataObjectsDeleted({ event, store }: EventContext & StoreContext): Promise<void> {
   const [, bagId, dataObjectIds] = new Storage.DataObjectsDeletedEvent(event).params
-  const dataObjects = await getDataObjectsInBag(store, bagId, dataObjectIds)
-  await Promise.all(dataObjects.map((o) => removeDataObject(store, o)))
+  const dataObjects = await getDataObjectsInBag(store, bagId, dataObjectIds, [
+    'videoThumbnail',
+    ...videoRelationsForCounters.map(item => `videoThumbnail.${item}`),
+    'videoMedia',
+    ...videoRelationsForCounters.map(item => `videoMedia.${item}`),
+  ])
+
+  await Promise.all(
+    dataObjects.map(async (dataObject) => {
+      // remember if video is fully active before update
+      const initialVideoActiveStatusThumbnail = dataObject.videoThumbnail
+        ? getVideoActiveStatus(dataObject.videoThumbnail)
+        : null
+      const initialVideoActiveStatusMedia = dataObject.videoMedia ? getVideoActiveStatus(dataObject.videoMedia) : null
```

Comment:
I'd reduce it to just 1 variable.
```suggestion
      const initialVideoActiveStatus = dataObject.videoThumbnail
        ? getVideoActiveStatus(dataObject.videoThumbnail)
        : dataObject.videoMedia
          ? getVideoActiveStatus(dataObject.videoMedia)
          : null
```

Reply:
Done. 

NOTE: This implementation relies on dataObject being associated to either thumbnail or video, but not both together. In current runtime implementation, that's always true (and will likely ever be).